src/Entity/Visitor.php

```diff
@@ -9,18 +9,29 @@
  */
 class Visitor {
 
-	private $totalImpressionCount = 0;
-	private $bucketIdentifier;
-	private $activeCategories;
+	private int $totalImpressionCount = 0;
+	private ?string $bucketIdentifier;
+	private array $activeCategories;
+	private ?int $minDisplayWidth;
```

Comment:
The visitor has only one concrete width, the actual width of their browser window.